library/core/src/marker.rs

```diff
@@ -469,6 +469,12 @@ pub macro Copy($item:item) {
 #[cfg_attr(not(test), rustc_diagnostic_item = "Sync")]
 #[lang = "sync"]
 #[rustc_on_unimplemented(
+    on(_Self = "std::cell::OnceCell<T>", note = "use `std::sync::OnceLock` instead"),
```

Comment:
Maybe a bit more wordy explanation will be better? Something along the lines of "You can try using `std::sync::OnceLock`, which implements `Sync`, instead"?

Reply:
sounds good